### PR TITLE
Added unit tests for multithreaded Get and Put usage. Used write_guard for Get

### PR DESCRIPTION
**Pull request by hmusta**

### CODE REVIEW

File: include/cache.hpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ class fixed_sized_cache
 
     Value Get(const Key &key) const
     {
-        read_guard lock{safe_op};
+        write_guard lock{safe_op};
```
Comment (karasikov) [inline include/cache.hpp]: why write_guard?

File: test/multithreading_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    #pragma omp parallel for num_threads(4)
+    for (size_t i = 0; i < 1000000; ++i) {
+        try {
+            cache.Get(i % 100);
+        } catch (...) {
+            cache.Put(i % 100, i);
+        }
+    }
+}
```
Comment (karasikov) [inline test/multithreading_tests.cpp]: run their clang-format for this file

Comment (karasikov) [reply test/multithreading_tests.cpp]: Here's a config https://github.com/karasikov/caches/blob/master/.clang-format

File: include/cache.hpp (hunk 1 above)
Comment (hmusta) [reply include/cache.hpp]: `Get` calls `Touch`, which may rearrange elements in the `cache_policy` object.

An alternative that would require more code would be to add mutexes to each policy class that this affects (LRU and LFU), then write custom copy constructors for them. Might be worth it

Comment (karasikov) [reply include/cache.hpp]: or you can pick between read and write guards, depending on the policy

Comment (karasikov) [reply include/cache.hpp]: but implementing mutexes in policies is better

**Review by karasikov — APPROVED** [review]

**Pull request merged**